gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagActionReminderScheduler.java

```diff
@@ -77,28 +95,45 @@ public DagActionReminderScheduler(StdSchedulerFactory schedulerFactory)
    * @throws SchedulerException
    */
   public void scheduleReminder(DagActionStore.LeaseParams leaseParams, long reminderDurationMillis,
-      boolean isDeadlineReminder)
-      throws SchedulerException {
+      boolean isDeadlineReminder) throws SchedulerException {
+    DagActionStore.DagAction dagAction = leaseParams.getDagAction();
     JobDetail jobDetail = createReminderJobDetail(leaseParams, isDeadlineReminder);
-    Trigger trigger = createReminderJobTrigger(leaseParams.getDagAction(), reminderDurationMillis,
+    Trigger trigger = createReminderJobTrigger(dagAction, reminderDurationMillis,
         System::currentTimeMillis, isDeadlineReminder);
-    log.info("Reminder set for dagAction {} to fire after {} ms, isDeadlineTrigger: {}",
-        leaseParams.getDagAction(), reminderDurationMillis, isDeadlineReminder);
-    quartzScheduler.scheduleJob(jobDetail, trigger);
+    log.info("Going to set reminder for dagAction {} to fire after {} ms, isDeadlineTrigger: {}",
+        dagAction, reminderDurationMillis, isDeadlineReminder);
+
+    try {
+      if (!this.deleteDeadlineDagActionCache.get(dagAction)) {
+        quartzScheduler.scheduleJob(jobDetail, trigger);
+      } else {
+        log.info("Ignoring {} because the delete equivalent of the same received already.", dagAction);
```

Comment:
`DagActionReminderScheduler` skipping adding reminder for action because already received delete for this action: {}.